src/chai-form.ts

```diff
@@ -45,6 +45,17 @@ export class ChaiForm extends LitElement {
     // initialized when the element is connected, if needed.
     this.flowInstanceId = localStorage.getItem('chai-flowInstanceId');
 
+    const hostName = window.location.hostname.split('.');
+    const tldOrLocalHost = hostName.pop();
+    const domainOrUndefined = hostName.pop();
+    let flowTypeLocalHostResilient;
+    if (domainOrUndefined == null) {
+      flowTypeLocalHostResilient = tldOrLocalHost ?? 'localhost';
+    } else {
+      flowTypeLocalHostResilient = domainOrUndefined + '.' + tldOrLocalHost;
```

Comment:
Please extract this logic to a helper function and add unit tests. Cover `localhost`, secure context IP address & port, apex domain (example.com), and one- and two-level subdomains.

Reply:
I have created a set of unit tests for this. 
One thing we do not have to worry about is a port in here. The documentation states, that "window.location.hostname" will return the hostname without the port. 
What does not actually work with this logic is an ip-address. It would interpret the sections of the ip-address as hostname and subdomains and would take the last to segments of the IP-Address. 
Question would be if we want to handle this. There is probably not going to be a real flow with an IP-address as identifier, so should we even detect if this looks like an ip-address? If we do decide to detect this should we throw an exception? 
Or just split it like a hostname and send that flow in, which will obviously fail later in the process?

Reply:
The case of domainOrUndefined being "." is not a concern from what I understand. 
if it is "." it will be caught by the "split(.)". If it is non-existent it will be correctly caught by "domainOrUndefined == null". 
I can't seem to find a way to create a flowType with a leading ".". Could you help me with an example domain to run the tests against that one?

Reply:
Thank you @WhiteHatTux! We don't need to handle IP addresses, that's my mistake -- I was thinking about the environment attribute, not the flowtype. I like that you added the test that documents the expected behavior for the IP address scenario. 🙂